plugin/core/sessions.py

```diff
@@ -1317,6 +1318,7 @@ def __init__(self, manager: Manager, logger: Logger, workspace_folders: list[Wor
         self._plugin: AbstractPlugin | None = None
         self._status_messages: dict[str, str] = {}
         self._semantic_tokens_map = get_semantic_tokens_map(config.semantic_tokens)
+        self._is_executing_refactoring_command = False
```

Comment:
I've only taken a first glance over the code in this PR but this kind of screams for a race condition. This value gets set before a request gets send and is reset when handling the response (I think). But the requests are async and there could be multiple open requests of this type at the same time. So this should probably be a local variable for each particular request instead.

Reply:
As long as it works in more cases I'm happy :)

I think it should be pretty natural to assume that server will handle the whole command and then send the response. It would probably have to get out of its way to do otherwise.